### ISSUE TITLE
Duplicating type definitions leads to the duplicated results

### ISSUE THREAD

**Issue by noskill** [trueagi-io/hyperon-experimental#458]
for example this program
```
        (: if (-> Bool Atom Atom $t))
        (: if (-> Bool Atom Atom $t))
        (: if (-> Bool Atom Atom $t))

        (= (test $V)
           (if $V a b)
        )


        !(test True)
```
gives this result:

[[a, a, a, a, a, a, a, a, a, a, a, a, a, a, a, a]]

I would expect the interpreter to give an error or a warning about duplicate definitions

**Comment by Necr0x0Der** [comment]
It is not a error by itself, because multiple type definitions (non-deterministic types) are ok, but  the idea of providing warnings with certain pragma settings is reasonable. The real issue here is multiple results, which is already raised:

https://github.com/trueagi-io/hyperon-experimental/issues/235

Not sure if this issue should be closed as duplication or its focus should be moved to the idea of having an option to raise warnings or errors for multiple type definitions with certain pragmas.

**Comment by vsbogd** [comment]
Closing it as behavior is fixed by fixing #235
